portals/publisher/src/main/webapp/source/src/app/components/Shared/PoliciesUI/AttachedPolicyCard.tsx

```diff
@@ -59,28 +60,30 @@ interface AttachedPolicyCardSharedProps {
     PolicyConfigurationEditDrawer: any;
 }
 
+// Option 1: `listOriginatedFromCommonPolicies` and `isApiRevision` are provided
+interface AttachedPolicyCardWithCommonProps extends AttachedPolicyCardSharedBaseProps {
+    listOriginatedFromCommonPolicies: string[];
+    isApiRevision: boolean;
+}
+
+// Option 2: Neither `listOriginatedFromCommonPolicies` nor `isApiRevision` are provided
+interface AttachedPolicyCardWithoutCommonProps extends AttachedPolicyCardSharedBaseProps {
+    listOriginatedFromCommonPolicies?: undefined;
+    isApiRevision?: undefined;
+}
+
+// Combine the two using a union type
+type AttachedPolicyCardSharedProps = AttachedPolicyCardWithCommonProps | AttachedPolicyCardWithoutCommonProps;
+
 /**
  * Renders a single sortable policy card.
  * @param {any} AttachedPolicyCardProps Input props from parent components.
  * @returns {TSX} Sortable attached policy card UI.
  */
-const AttachedPolicyCardShared: FC<AttachedPolicyCardSharedProps> = ({
-    policyObj,
-    currentFlow,
-    verb,
-    target,
-    allPolicies,
-    isAPILevelPolicy,
-    drawerOpen,
-    handleDrawerOpen,
-    handlePolicyDownload,
-    handleDelete,
-    setDrawerOpen,
-    PolicyConfigurationEditDrawer
-}) => {
+const AttachedPolicyCardShared: FC<AttachedPolicyCardSharedProps> = (props) => {
```

Comment:
Shall we revert back to the prop destructuring here?

```suggestion
const AttachedPolicyCardShared: FC<AttachedPolicyCardSharedProps> = ({
     ...
}) => {
```

Reply:
This change was introduced based on the following considerations:

Data flows through this shared component to both the Global Policy UIs and the API/Common Policy UIs. In the context of Global Policy flows, the concept represented by listOriginatedFromCommonPolicies does not apply, and as a result, this prop is not passed. Furthermore, in the API/Common Policy flows, there are scenarios where listOriginatedFromCommonPolicies may not exist.

Due to the lack of a definitive way to distinguish between a Global Policy page and a standard policy page based solely on the **value** of listOriginatedFromCommonPolicies in the props, destructuring was removed. Instead, a check was introduced to verify **whether the corresponding prop exists among the passed props**, rather than relying on its value.